### PR TITLE
Add api mode

### PR DESCRIPTION
**Pull request by BatuhanW**
- Added apiMode to SAO Config for skipping certain steps on api mode.
- Added api.ts to use with API
- Updated SAOFile

### CODE REVIEW

File: src/Helper/prettier/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import path from "path";
+import fs from "fs";
+import { format } from "prettier";
```
Comment (aliemir) [inline src/Helper/prettier/index.ts:3]: Prettier is left in the `devDependencies`

Comment (BatuhanW) [reply src/Helper/prettier/index.ts:3]: I moved it 🚀 package-lock.json is collapsed by default, you can see the change there.

File: src/cli.ts [hunk 2]
```diff
@@ -189,11 +187,12 @@ const cli = async (): Promise<void> => {
 
     const sao = new SAO({
         generator,
-        outDir: finalProjectDir,
+        outDir: process.cwd(),
```
Comment (aliemir) [inline src/cli.ts:190]: Can we try this with `npm run build:global` (with global install) and running the executable from the globally installed one? 🤔

**Comment by aliemir** [comment]
Should we skip package install at `completed` in `saofile.ts`? @BatuhanW @omeraplak

**Comment by BatuhanW** [comment]
> Should we skip package install at `completed` in `saofile.ts`? @BatuhanW @omeraplak

Yeah, we already do it: https://github.com/pankod/superplate/pull/327/files#diff-d49e14c45125538cf0fc5cdcdbb6b85c9e189657ca11bc6cecd504c43a8baaedR347

**Review by aliemir — APPROVED** [review]

**Review by omeraplak — APPROVED** [review]

**Pull request merged**